src/indicators/FoodIndicator.js

```diff
@@ -1,56 +1,39 @@
-import React, { createRef, useEffect, useRef, useState } from "react";
+import React, { useEffect } from "react";
 import style from "../App.module.css";
+import { Progress } from "antd";
+import "antd/dist/antd.css";
+import { useInterval } from "usehooks-ts";
+
+const ProgressLine = ({ foodPercent }) => (
+  <>
+    <Progress
+      strokeWidth={15}
+      strokeColor={{
+        "0%": "red",
+        "100%": "green",
+      }}
+      percent={foodPercent}
+    />
+  </>
+);
 
 export const FoodIndicator = ({
   foodPercent,
   setFoodPercent,
-  buttonFeed,
-  setButtonFeedEntered,
+  stateIndicator,
 }) => {
-  const getWidthEl = () => {
-    const whiteBlockEl = document.getElementById("whiteBlock");
-    let widthWhiteBlockEl = window
-      .getComputedStyle(whiteBlockEl)
-      .getPropertyValue("width");
-
-    return widthWhiteBlockEl;
-  };
-
-  useEffect(() => {
-    const interval = setInterval(() => {
-      let widthEl = getWidthEl();
-
-      if (widthEl === "100px" || foodPercent === 0 || buttonFeed === true) {
-        setButtonFeedEntered(false);
-        return () => clearInterval(interval);
-      } else {
-        setFoodPercent(foodPercent - 1);
-      }
-    }, 495);
-
-    return () => clearInterval(interval);
-  });
+  useInterval(
+    () => {
+      setFoodPercent(foodPercent - 1);
+    },
+    foodPercent === 0 || stateIndicator === "sick" || stateIndicator === "dead"
```

Comment:
such complex conditions should be extracted into their own variable, like `shouldDecreaseFoodLevel`

Reply:
fixed